engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -207,8 +212,24 @@ private void initMaterials() {
     private void initRenderGraph() {
         // FIXME: init pipeline without specifying them as a field in this class
         NodeFactory nodeFactory = new NodeFactory(context);
+
+        // ShadowMap generation
+        FBOConfig shadowMapConfig =
+                new FBOConfig(ShadowMapNode.SHADOW_MAP, FBO.Type.NO_COLOR, shadowMapResolutionDependentFBOs).useDepthBuffer();
+        BufferClearingNode.RequiredData shadowMapClearingData =
+                new BufferClearingNode.RequiredData(shadowMapConfig, GL_DEPTH_BUFFER_BIT);
+        Node shadowMapClearingNode = nodeFactory.createInstance(BufferClearingNode.class, shadowMapClearingData);
```

Comment:
It felt neater to put RequireData as a class within the Node as it is very node-specific.  But I can certainly put all the RequireData classes into dag/nodes/requiredData, it would certainly shorten things in this part of the code.

What do you mean with "it all looks public anyway"?